### PR TITLE
openvswitch: add geneve, gre, vxlan tunnelling support

### PR DESCRIPTION
**Pull request by yousong**
Maintainer: @commodo 
Compile tested: mips_24kc, qemu malta, r6329+1-83d8df1
Run tested: mips_24kc, qemu malta, r6329+1-83d8df1, daemon runs, ovs-vsctl works

### CODE REVIEW

File: net/openvswitch/Makefile [hunk 1]
```diff
@@ -156,12 +156,12 @@ CONFIGURE_VARS += \
 	ovs_cv_flake8=no \
 	ovs_cv_python3=no \
 	ovs_cv_sphinx=no \
+	ovs_cv_python=$(PYTHON) \
```
Comment (commodo) [inline net/openvswitch/Makefile]: why not use `$(HOST_PYTHON_BIN)` here too ?
`$(PYTHON)` var resolves to `python` ; which is the host-system's Python ; maybe it's a good idea to use the build-system's Python executable here as well;

or, is this variable used for the target's Python ?

**Comment by commodo** [comment]
LGTM :)
Thanks for taking over OVS.

I still feel that maybe OVS kmods should not be packaged into core ; I am referring to `kmod-geneve` here.

File: net/openvswitch/Makefile (hunk 1 above)
Comment (yousong) [reply net/openvswitch/Makefile]: Build system of Open vSwitch does not differentiate host and target python, but it replaces all `@PYTHON@` in the code to be run on target.  That's why I chose to patch `run_python` to use the newly introduced `PYTHON_HOST` instead for the build process.

Comment (commodo) [reply net/openvswitch/Makefile]: ack ;
thanks for the info

**Comment by neheb** [comment]
The website is available over HTTPS. Probably good idea to update the URLs.

**Comment by yousong** [comment]
Hi @neheb , Thank you for the heads-up but I guess better not in this case.  Here are the observations leading to this conclusion

 - The official website suggests "http://openvswitch.org"
 - www.openvswitch.org and openvswitch.org point to different ip addresses
 - "https://openvswitch.org" uses a self-signed certificate and the corresponding https download URL fails with "No data received"

**Comment by neheb** [comment]
Weird. openvswitch.org redirects to www.openvswitch.org for me.

https://openvswitch.org is indeed broken. but the www. version works. Probably misconfigured website.

**Comment by yousong** [comment]
Hi @neheb , I missed the redirection from openvswitch.org to www.openvswitch.org.  Now let's switch to https://www.openvswitch.org for good ;)

File: net/openvswitch/Makefile [hunk 2]
```diff
@@ -102,7 +102,7 @@ endef
 define Package/openvswitch-python
   $(call Package/openvswitch/Default)
   TITLE:=Open vSwitch Python Support
-  DEPENDS:=+openvswitch +PACKAGE_openvswitch-python:python +PACKAGE_openvswitch-python:python-six
+  DEPENDS:=+python +python-six
```
Comment (commodo) [inline net/openvswitch/Makefile]: so, according to the discussion via email with Felix, drop this commit ?

**Comment by commodo** [comment]
@yousong 
i forgot to say; 
LGTM :)

fine from my side

File: net/openvswitch/Makefile (hunk 2 above)
Comment (yousong) [reply net/openvswitch/Makefile]: Yes, I just reworked this to only remove openvswitch as a dependency while retaining the conditional deps

Comment (commodo) [reply net/openvswitch/Makefile]: ack

**Comment by yousong** [comment]
v2 <- v1

 - Fixup removing `SUPPORTED_KERNELS` by also remove it from kmod-openvswitch depends
 - Revert removing conditional part in depends definition of openvswitch-python

**Pull request merged**